anndata/core/sparsedataset.py

```diff
@@ -8,79 +8,117 @@
 
 # TODO:
 # - think about supporting the COO format
-from collections.abc import Iterable
+import collections.abc as cabc
 from itertools import accumulate, chain
-from typing import Union, NamedTuple, Tuple, Sequence, Type
+from typing import Union, NamedTuple, Tuple, Sequence, Iterable, Type
 from warnings import warn
 
 import h5py
 import numpy as np
 import scipy.sparse as ss
 from scipy.sparse import _sparsetools
 
-from ...utils import unpack_index
+try:
+    # Not really important, just for IDEs to be more helpful
+    from scipy.sparse.compressed import _cs_matrix
+except ImportError:
+    _cs_matrix = ss.spmatrix
+
+from .index import unpack_index, Index
 
 
 class BackedFormat(NamedTuple):
     format_str: str
-    backed_type: Type['BackedSparseMatrixMixin']
+    backed_type: Type['BackedSparseMatrix']
     memory_type: Type[ss.spmatrix]
 
 
-class BackedSparseMatrixMixin:
+class BackedSparseMatrix(_cs_matrix):
     """\
     Mixin class for backed sparse matrices.
 
     Largely needed for the case `backed_sparse_csr(...)[:]`, since that calls
     copy on `.data`, `.indices`, and `.indptr`.
     """
 
-    def copy(self):
+    def copy(self) -> ss.spmatrix:
         if isinstance(self.data, h5py.Dataset):
-            return SparseDataset(self.data.parent).tomemory()
+            return SparseDataset(self.data.parent).to_memory()
         else:
             return super().copy()
 
+    def _set_many(self, i: Iterable[int], j: Iterable[int], x):
```

Comment:
No, they have been like this because that’s the way they were defined and attached to the class in scipy. The lines below like `backed_csr_matrix._zero_many = _zero_many` mean that the only difference to defining them like this was a different `__qualname__` and one more indentation level.